process/block/metablock.go

```diff
@@ -868,28 +922,34 @@ func (mp *metaProcessor) createEpochStartBody(metaBlock *block.MetaBlock) (data.
 		}
 	}
 
-	metaBlock.EpochStart.Economics.RewardsForProtocolSustainability.Set(mp.epochRewardsCreator.GetProtocolSustainabilityRewards())
+	return rewardMiniBlocks, nil
+}
 
-	err = mp.epochSystemSCProcessor.ProcessDelegationRewards(rewardMiniBlocks, mp.epochRewardsCreator.GetLocalTxCache())
-	if err != nil {
-		return nil, err
+func (mp *metaProcessor) createRewardsMiniBlocks(
+	metaHeader data.MetaHeaderHandler,
+	allValidatorsInfo state.ShardValidatorsInfoMapHandler,
+	computedEconomics *block.Economics,
+) (block.MiniBlockSlice, error) {
+	if !metaHeader.IsHeaderV3() {
+		return mp.epochRewardsCreator.CreateRewardsMiniBlocks(metaHeader, allValidatorsInfo, computedEconomics)
 	}
 
-	validatorMiniBlocks, err := mp.validatorInfoCreator.CreateValidatorInfoMiniBlocks(allValidatorsInfo)
-	if err != nil {
-		return nil, err
+	epochRewardsCreatorV3, ok := mp.epochRewardsCreator.(process.RewardsCreatorHeaderV3)
```

Comment:
Why have this cast here and not simply change internal `epochRewardsCreator` interface to be of type `RewardsCreatorHeaderV3` ?